app/notebook/server/CalcWebSocketService.scala

```diff
@@ -43,6 +45,9 @@ class CalcWebSocketService(
     var calculator: ActorRef = null
     var wss: List[WebSockWrapper] = Nil
 
+    protected val notebookStartTime = LocalTime.now()
+    private var lastCellExecutionTime: Option[LocalTime] = None
```

Comment:
well timeout is kinda configurable.
so if user used `Lock Screen` (instead of suspend), he should see the results and they'd be autosaved (just will need to restart the kernel).